Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -23,6 +25,11 @@ public static int IndexOfFirstMismatch(this string value, string expected, Strin
             return -1;
         }
 
+        private static Func<char, char, bool> GetComparer(StringComparison stringComparison) =>
```

Comment:
I'm surprised that all this funciness is faster. It _is_ a bit more difficult to read.

Reply:
`GetComparer` lazy initializes and caches the two `Func`s in static fields.

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJ9KBZATwGENopKBvUy7ygel4BUXHpQAOCNADcI8agQBs1AKwAeAMYALCMkqbtSSsAD2RjAD5KAcTgxGRsKO2IAFMdOU0AcwB2RhHEZoOABKSgBeM2ERaI8fPwCgqJjkgH5KZwAPA3pQiN0tBAA6ABUjAFVRUUQASW9pCQhvGEzcsPztEvLKmrrtNEbmnKTk6JB0nFU9HSmDN3Ng50zs3IsM8LacgG5SQSTxKRk4OUUJ9QKDGcMTc0o/ABM0bwgMcPSsyhzw1fWP7ZIRfbSWQ4BTKM76do6OYWe6PZ7VOL+NRBV5LD4rSGdCpVBC1er9JotH5TLHdXG9BqErZJPYSIFHEEnSbnSGza4Way2eyOfwIVzXWK+fyBKAhL7DHheIUJUUSmJpWFPDAI6XI2X/EY8MaK55/bgCXjcJIUagAFjotFozmCSU4Gpi9V03KcCFenLsDhdzil8RFcDGMAQAFcQnqeABfUjhoA=

So it's equivalent to this modulo the lazy initialization.
```c#
private static Func<char, char, bool> ordinal = (x, y) => x == y;
private static Func<char, char, bool> ordinalIgnorecase = (x, y) => char.ToUpperInvariant(x) == char.ToUpperInvariant(y);
    
private static Func<char, char, bool> GetComparer(bool ignoreCase) =>
    ignoreCase ? ordinalIgnorecase : ordinal;
```

Reply:
Maybe call is GetCharComparer? Because I was confusing it with the regular StringComparer.